nav2_behavior_tree/plugins/action/truncate_path_action.cpp

```diff
@@ -31,14 +31,20 @@ TruncatePath::TruncatePath(
   const std::string & name,
   const BT::NodeConfiguration & conf)
 : BT::ActionNodeBase(name, conf),
-  distance_(1.0)
+  distance_(1.0),
+  first_time_(true)
 {
-  getInput("distance", distance_);
 }
 
 inline BT::NodeStatus TruncatePath::tick()
 {
   setStatus(BT::NodeStatus::RUNNING);
+  getInput("distance", distance_);
```

Comment:
You get distance here -- no need for the first time stuff if you're getting it just a line above every iteration

Reply:
done